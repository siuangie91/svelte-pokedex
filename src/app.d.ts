/// <reference types="@sveltejs/kit" />

// See https://kit.svelte.dev/docs/types#app
// for information about these interfaces
// and what to do when importing types
declare namespace App {
  // interface Locals {}
  // interface Platform {}
  // interface Session {}
  // interface Stuff {}
}

declare namespace PokemonAPI {
  interface PokemonSpecies {
    name: string;
    url: string;
  }

  interface Entry {
    entry_number: number;
    pokemon_species: PokemonSpecies;
  }

  interface Generation {
    pokemon_entries: Entry[];
  }

  interface Sprites {
    other: {
      dream_world: {
        front_default: string;
      };
    };
  }

  interface Type {
    type: {
      slot: number;
      name: string;
    };
  }

  interface Pokemon {
    name: string;
    species: PokemonSpecies;
    sprites: Sprites;
    types: Type[];
  }
}
