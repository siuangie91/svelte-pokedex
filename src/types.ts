interface Identifiers {
  id: number;
  name: string;
}

export interface Specy {
  pokemon: {
    id: number;
    name: string;
    forms: {
      sprites: {
        sprites: string;
      }[];
    }[];
  }[];
}

export interface Generation1 {
  gen1: {
    name: string;
    species: Specy[];
  };
}

export interface Type {
  type: {
    name: string;
  };
}
export interface PokemonStats {
  pokemon: {
    id: number;
    name: string;
    types: Type[];
    species: {
      descriptions: {
        id: number;
        text: string;
      }[];
      evolutionChain: {
        chain: Identifiers[];
      };
    };
    images: {
      sprites: string;
    }[];
  }[];
}

export interface Summary extends Identifiers {
  image: string;
}

export interface PokemonLookup {
  [key: Summary['name']]: Summary;
}

export interface TypeColorLookup {
  [key: string]: {
    color: string;
    dark: boolean;
  };
}
