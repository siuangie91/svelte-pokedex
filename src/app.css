@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  html {
    background: url(http://pngimg.com/uploads/pokeball/pokeball_PNG8.png) no-repeat;
    background-position: top center;
    background-size: 100vh;
    min-height: 100vh;
  }
  body {
    /* position: relative; */
    min-height: 100vh;
    background: rgba(255, 255, 255, 0.5);
    display: flex;
    flex-direction: column;
  }
  h1 {
    @apply my-4;
    @apply text-3xl;
    @apply font-bold;
  }
  h2 {
    @apply my-3;
    @apply text-xl;
    @apply font-bold;
  }
}
