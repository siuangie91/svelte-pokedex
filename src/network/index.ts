// TODO test
import axios from 'axios';

export const POKEAPI_GQL_URL = 'https://beta.pokeapi.co/graphql/v1beta';

export const fetchGraphQL = async <ResponseShape>(
  query: string,
  variables: {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    [key: number | string]: any;
  },
  operationName: string,
) => {
  try {
    const { data } = await axios.post<ResponseShape>(POKEAPI_GQL_URL, {
      query,
      variables,
      operationName,
    });

    return data;
  } catch (error) {
    console.error(error);
  }
};
