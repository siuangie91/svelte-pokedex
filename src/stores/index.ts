import { writable } from 'svelte/store';
import type { PokemonLookup } from 'src/types';

export const seenPokemon = writable<PokemonLookup>({});
