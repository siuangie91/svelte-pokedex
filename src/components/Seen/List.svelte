<script lang="ts">
  import { seenPokemon } from 'src/stores';
  import type { PokemonLookup } from 'src/types';
  import { capitalizeFirstLetter } from 'utils';

  let pokemons: PokemonLookup = {};

  seenPokemon.subscribe(seen => {
    if (!seen) return;
    pokemons = { ...seen };
  });

  $: pokemonsList = Object.values(pokemons).sort((a, b) => a.id - b.id);
</script>

{#if !pokemonsList.length}
  <p>You haven't seen any Pokémon yet!</p>
{:else}
  <ul>
    {#each pokemonsList as { id, name, image }}
      <li>
        <p>
          <a href={`/pokemon/${name}`} target="_blank">
            <span>{id}. {capitalizeFirstLetter(name)}</span>
            <img src={image} alt={name} />
            <span class="col-span-1">&#10548;</span><!-- ⤴ -->
          </a>
        </p>
      </li>
    {/each}
  </ul>
{/if}

<style lang="postcss">
  li {
    @apply rounded;
    @apply bg-white/80;
    @apply px-2;
    @apply py-1;
    @apply mb-2;

    @apply shadow-sm;
    @apply shadow-slate-500;
    @apply hover:shadow-none;
  }

  a {
    @apply grid;
    @apply grid-cols-10;
    @apply sm:grid-cols-12;

    /* @apply text-cyan-800;
    @apply underline;
    @apply hover:no-underline; */
  }

  span {
    @apply inline-flex;
    @apply items-center;

    @apply first-of-type:col-span-7;
    @apply sm:first-of-type:col-span-9;

    @apply last-of-type:col-span-1;
  }

  img {
    @apply h-14;

    @apply col-span-2;
  }
</style>
