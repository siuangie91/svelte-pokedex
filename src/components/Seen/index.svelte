<script lang="ts">
  import type { PokemonLookupGQL } from 'src/types';
  import Container from 'components/shared/Container.svelte';
  import List from './List.svelte';

  export let entries: PokemonLookupGQL = {};
</script>

<Container title="Here's what you've seen so far:">
  <List pokemons={entries} />
</Container>
