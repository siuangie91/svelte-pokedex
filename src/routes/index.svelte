<script lang="ts">
  import type { PokemonLookup } from 'src/types';
  import Selection from 'components/Selection/index.svelte';
  import Seen from 'components/Seen/index.svelte';

  const entries: PokemonLookup = {};

  // TODO test
  const handleAdd = (event: CustomEvent) => {
    const entry = event.detail;
    console.log('event', event.detail);
    const { name } = entry;

    if (!entries[name]) {
      entries[name] = entry;
    }

    // TODO delete
    console.log('entries', entries);
  };
</script>

<svelte:head>
  <title>Svelte Pokédex</title>
</svelte:head>

<h1>Svelte Pokédex</h1>
<div>
  <Selection entryAddedHandler={handleAdd} />
  <Seen {entries} />
</div>

<style lang="postcss">
  h1 {
    @apply text-center;
  }

  @media (min-width: 640px) {
    div {
      @apply grid;
      @apply grid-cols-2;
      @apply gap-4;
      min-height: 50vh;
    }
  }
</style>
