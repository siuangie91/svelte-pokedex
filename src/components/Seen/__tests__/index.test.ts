import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';
import { seenPokemon } from 'src/stores';
import Seen from '../index.svelte';

describe('Card', () => {
  it('renders a list of seen pokemon', () => {
    const pokemons = {
      fake1: {
        id: 1,
        name: 'Fake1',
        image: 'path/to/one'
      },
      fake2: {
        id: 2,
        name: 'Fake2',
        image: 'path/to/two'
      },
      fake3: {
        id: 3,
        name: 'Fake3',
        image: 'path/to/three'
      },
    };

    seenPokemon.set(pokemons);

    const { container } = render(Seen);

    expect(container).toMatchSnapshot();
  });
});
