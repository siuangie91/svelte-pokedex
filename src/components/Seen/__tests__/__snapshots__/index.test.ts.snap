// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders a list of seen pokemon 1`] = `
<body>
  <div>
    <section
      class="svelte-142rirh"
    >
      <h2>
        Here's what you've seen so far:
      </h2>
       
      <ul>
        <li
          class="svelte-gtgtc4"
        >
          <p>
            <a
              class="svelte-gtgtc4"
              href="/pokemon/Fake1"
              target="_blank"
            >
              <span
                class="id-name svelte-gtgtc4"
              >
                1
                . 
                Fake1
              </span>
               
              <img
                alt="Fake1"
                class="svelte-gtgtc4"
                src="path/to/one"
              />
               
              <span
                aria-hidden="true"
                class="link-icon svelte-gtgtc4"
              >
                <span
                  class="svelte-gtgtc4"
                >
                  ☐
                </span>
                 
                <span
                  class="link-icon-arrow svelte-gtgtc4"
                >
                  ↗
                </span>
              </span>
            </a>
          </p>
           
        </li>
        <li
          class="svelte-gtgtc4"
        >
          <p>
            <a
              class="svelte-gtgtc4"
              href="/pokemon/Fake2"
              target="_blank"
            >
              <span
                class="id-name svelte-gtgtc4"
              >
                2
                . 
                Fake2
              </span>
               
              <img
                alt="Fake2"
                class="svelte-gtgtc4"
                src="path/to/two"
              />
               
              <span
                aria-hidden="true"
                class="link-icon svelte-gtgtc4"
              >
                <span
                  class="svelte-gtgtc4"
                >
                  ☐
                </span>
                 
                <span
                  class="link-icon-arrow svelte-gtgtc4"
                >
                  ↗
                </span>
              </span>
            </a>
          </p>
           
        </li>
        <li
          class="svelte-gtgtc4"
        >
          <p>
            <a
              class="svelte-gtgtc4"
              href="/pokemon/Fake3"
              target="_blank"
            >
              <span
                class="id-name svelte-gtgtc4"
              >
                3
                . 
                Fake3
              </span>
               
              <img
                alt="Fake3"
                class="svelte-gtgtc4"
                src="path/to/three"
              />
               
              <span
                aria-hidden="true"
                class="link-icon svelte-gtgtc4"
              >
                <span
                  class="svelte-gtgtc4"
                >
                  ☐
                </span>
                 
                <span
                  class="link-icon-arrow svelte-gtgtc4"
                >
                  ↗
                </span>
              </span>
            </a>
          </p>
           
        </li>
      </ul>
      
    </section>
  </div>
</body>
`;
